src/main/java/frc/robot/subsystems/snapback/shooter/ShooterIOSim.java

```diff
@@ -1,3 +1,105 @@
 package frc.robot.subsystems.snapback.shooter;
 
-public class ShooterIOSim {}
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import frc.robot.constants.Constants;
+
+public class ShooterIOSim implements ShooterIO {
+
+  private DCMotorSim leftFlywheelMotorSim;
+  private DCMotorSim rightFlywheelMotorSim;
+  private DCMotorSim acceleratorMotorSim;
+
+  private double leftFlywheelMotorAppliedVolts = 0.0;
+  private double rightFlywheelMotorAppliedVolts = 0.0;
+  private double acceleratorMotorAppliedVolts = 0.0;
+
+  private double leftFlywheelVelocitySetpoint = 0.0;
+  private double rightFlywheelVelocitySetpoint = 0.0;
+
+  public ShooterIOSim() {
+    leftFlywheelMotorSim =
+        new DCMotorSim(
+            LinearSystemId.createDCMotorSystem(
+                ShooterConstants.FLYWHEEL_GEARBOX, 0.004, ShooterConstants.FLYWHEEL_GEAR_REDUCTION),
+            ShooterConstants.FLYWHEEL_GEARBOX,
+            0.004);
+    rightFlywheelMotorSim =
+        new DCMotorSim(
+            LinearSystemId.createDCMotorSystem(
+                ShooterConstants.FLYWHEEL_GEARBOX, 0.004, ShooterConstants.FLYWHEEL_GEAR_REDUCTION),
+            ShooterConstants.FLYWHEEL_GEARBOX,
+            0.004);
+    acceleratorMotorSim =
+        new DCMotorSim(
+            LinearSystemId.createDCMotorSystem(
+                ShooterConstants.ACCELERATOR_GEARBOX, 0.004, ShooterConstants.ACCELERATOR_GEAR_REDUCTION),
+            ShooterConstants.ACCELERATOR_GEARBOX,
+            0.004);
+  }
+
+  @Override
+  public void updateInputs(ShooterIOInputs inputs) {
+
+    leftFlywheelMotorSim.update(Constants.LOOP_PERIOD_SECONDS);
+    rightFlywheelMotorSim.update(Constants.LOOP_PERIOD_SECONDS);
+    acceleratorMotorSim.update(Constants.LOOP_PERIOD_SECONDS);
+
+    leftFlywheelMotorSim.setInputVoltage(leftFlywheelMotorAppliedVolts);
```

Comment:
Because we are setting the voltage in each method, we don't need to set the voltage in update inputs:
```suggestion
```